pkg/skaffold/util/port.go

```diff
@@ -129,16 +138,20 @@ func GetAvailablePort(port int, usedPorts *PortSet) int {
 	return p
 }
 
-func getPortIfAvailable(p int, usedPorts *PortSet) bool {
+func getPortIfAvailable(address string, p int, usedPorts *PortSet) bool {
 	if alreadySet := usedPorts.LoadOrSet(p); alreadySet {
 		return false
 	}
 
-	return IsPortFree(p)
+	return IsPortFreeWithAddress(address, p)
 }
 
 func IsPortFree(p int) bool {
-	l, err := net.Listen("tcp", fmt.Sprintf(":%d", p))
+	return IsPortFreeWithAddress(Any, p)
+}
+
+func IsPortFreeWithAddress(address string, p int) bool {
+	l, err := net.Listen("tcp", fmt.Sprintf("%s:%d", address, p))
 	if err != nil {
 		return false
 	}
```

Comment:
So this fix isn't quite enough as it still leaves the previous situation, where a local port may be available but the INADDR_ANY:port isn't available.  We should test both conditions.   Here's my suggestion:
```suggestion
	// Ensure the port is available across all interfaces
	if l, err := net.Listen("tcp", fmt.Sprintf(":%d", p));  err != nil {
		return false
	} else {
		l.Close()
	}
	if address != Any {
		// Ensure the port is available on the specific interface too
		if l, err := net.Listen("tcp", fmt.Sprintf("%s:%d", address, p));  err != nil {
			return false
		} else {
			l.Close()
		}
	}
```